harness/determined/common/declarative_argparse.py

```diff
@@ -121,7 +121,7 @@ class BoolOptArg(NamedTuple):
     true_name: str
     false_name: str
     dest: str
-    default: bool = False
+    default: Optional[bool] = None
```

Comment:
One downside of the new change is that its defaults make the `--help` look weird:
```
usage: det user edit [-h] [--display-name DISPLAY_NAME] [--username USERNAME] [--remote] [--no-remote] [--activate] [--deactivate] [--admin] [--no-admin] target_user

positional arguments:
  target_user           name of user that should be edited

optional arguments:
  -h, --help            show this help message and exit
  --display-name DISPLAY_NAME
                        new display name for target_user (default: None)
  --username USERNAME   new username for target_user (default: None)
  --remote              set as remote user (default: None)
  --no-remote           set as non-remote user
  --activate            set user as active (default: None)
  --deactivate          deactivate user
  --admin               grant user admin permissions (default: None)
  --no-admin            remove admin permissions from user
```

Nothing comes immediately to mind immediately for an optimal message. Suppressing or modifying the default _might_ be enough?

And I guess while this thread is open, I think there's an opportunity to massage the help text a little so it makes sense explicitly in this help message (like if the boolean pairs could be visually matched, or if it could be obvious that they set opposite values to the same attribute).

Reply:
As per above comment. Maybe we want to do these changes in a separate PR